src/vcpkg/binarycaching.cpp

```diff
@@ -231,15 +233,16 @@ namespace
                 std::error_code ec;
                 if (can_attempt_rename)
                 {
-                    // if we are already on the same filesystem we can just rename into place
                     m_fs.rename_or_delete(zip_path, archive_path, ec);
```

Comment:
Before #802, files were copied to the binary cache, meaning the original file in zip_path did still exist after this function ended. However, this file is gone with `rename_or_delete()`